### PR TITLE
Monte-carlo pi estimation

### PR DESCRIPTION
**Pull request by tremmelg**

### CODE REVIEW

File: examples/tests/monte-carlo.cpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+}
+}
+#endif
+
+struct estimate_pi {
+  float operator()(unsigned int thread_id) const {
+    float sum = 0;
+    unsigned int N = 5000; // samples per stream
+
+    // Workaround
```
Comment (Ruyk) [inline examples/tests/monte-carlo.cpp]: workaround for what?

File: examples/tests/monte-carlo.cpp [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    nextafterf(1.f,2.f);
+    logl(1.f);
+    sqrtf(1.f);
+    logf(1.f);
+
+    // create a random number generator
+    std::default_random_engine rng(thread_id);
+
+    // create a mapping from random numbers to [0,1)
+    std::uniform_real_distribution<float> u01(0, 1);
```
Comment (Ruyk) [inline examples/tests/monte-carlo.cpp]: is this working on the GPU device? I am surprised

**Review by Ruyk — CHANGES_REQUESTED** [review]
Looks you have done an interesting workaround to get random numbers on the device, can you elaborate a bit on the comments on how is this implemented?

File: examples/tests/monte-carlo.cpp (hunk 2 above)
Comment (tremmelg) [reply examples/tests/monte-carlo.cpp]: Not right out of the box, but yes. @DuncanMcBain came up with the tweaks to make it work.

**Review by Ruyk — APPROVED** [review]

**Comment by Ruyk** [comment]
LGTM

**Pull request merged**